doc/source/scripts/gen_format_overview_classes.py

```diff
@@ -7,115 +7,167 @@
     ("Converter", _CONVERTERS),
-):
-    for fmt, klass in dct.items():
-        if fmt in ("CHAIN", "MEMORY", "MINIMAL", "NULL"):
+]
+
+FILE_TYPES: dict[str, dict[HANDLER_T, Any]] = defaultdict(dict)
+
+for handler, dct in handlers:
+    for format, klass in dct.items():
+        if format in {"CHAIN", "MEMORY", "MINIMAL", "NULL"}:
             continue  # get their own pages
-        FILE_TYPES[fmt][clstype] = klass
+        FILE_TYPES[format][handler] = klass
+
 
 sorted_types = sorted(FILE_TYPES.items())
 
-SUCCESS = "\u2713"
+SUCCESS = "\u2713"  # checkmark
 FAIL = ""
 
 
-class FormatOverview(TableWriter):
-    filename = "formats/format_overview.txt"
-    include_table = "Table of all supported formats in MDAnalysis"
-    preprocess = ["keys"]
-    headings = ["File type", "Description", "Topology", "Coordinates", "Read", "Write"]
-
-    def _set_up_input(self):
-        return sorted_types
-
-    def _file_type(self, fmt, handlers):
-        return self.sphinx_ref(fmt, self.keys[-1], suffix="-format")
+def _create_key(fmt: str, handlers: dict[HANDLER_T, Type[Any]]) -> str:
+    if fmt in DESCRIPTIONS:
+        key = fmt
+    else:
+        key = list(handlers.values())[0].format[0]
 
-    def _keys(self, fmt, handlers):
-        if fmt in DESCRIPTIONS:
+        # raise an informative error
+        if key not in DESCRIPTIONS:
             key = fmt
-        else:
-            key = list(handlers.values())[0].format[0]
+    return key
 
-            # raise an informative error
-            if key not in DESCRIPTIONS:
-                key = fmt
-        return key
 
-    def _description(self, fmt, handlers):
-        return DESCRIPTIONS[self.keys[-1]]
+def _file_type(fmt: str, handlers: dict[HANDLER_T, Type[Any]], key: str) -> str:
+    return base.sphinx_ref(txt=fmt, label=key, suffix="-format")
 
-    def _topology(self, fmt, handlers):
-        if "Topology parser" in handlers:
-            return SUCCESS
-        return FAIL
-
-    def _coordinates(self, fmt, handlers):
-        if "Coordinate reader" in handlers:
-            return SUCCESS
-        return FAIL
 
-    def _read(self, fmt, handlers):
-        return SUCCESS
-
-    def _write(self, fmt, handlers):
-        if "Coordinate writer" in handlers:
-            return SUCCESS
-        if "Converter" in handlers:
-            return SUCCESS
-        return FAIL
+def _description(fmt: str, handlers: dict[HANDLER_T, Type[Any]], key: str) -> str:
+    return DESCRIPTIONS[key]
 
 
-class CoordinateReaders(FormatOverview):
-    filename = "formats/coordinate_readers.txt"
-    include_table = "Table of supported coordinate readers and the information read"
-    headings = ["File type", "Description", "Velocities", "Forces"]
+class FormatOverview:
+    def __init__(self) -> None:
+        def _topology(fmt: str, handlers: dict[HANDLER_T, Type[Any]], key: str) -> str:
+            if "Topology parser" in handlers:
+                return SUCCESS
+            return FAIL
```

Comment:
```suggestion
            return SUCCESS if "Topology parser" in handlers else FAIL
```

Maybe a bit shorter? Just an aesthetic suggestion.